packages/analytics/analytics-chart/src/components/SparklineChart.vue

```diff
@@ -0,0 +1,384 @@
+    <div
+      v-if="!hasData"
+      class="no-data"
+      data-testId="sparkline-empty"
+    >
+      <slot name="empty">
+        {{ i18n.t('sparkline_no_data') }}
+      </slot>
+    </div>
+    <Bar
+      v-else-if="type === 'bar'"
+      :chart-id="chartId"
+      :data="chartData"
+      data-testId="sparkline-bar"
+      :options="(options as any)"
+      :plugins="plugins"
+    />
+    <Line
+      v-else-if="type === 'line' || type === 'step'"
+      :chart-id="chartId"
+      :data="chartData"
+      :data-testId="type === 'line' ? 'sparkline-line' : 'sparkline-step'"
+      :options="(options as any)"
+      :plugins="plugins"
+    />
+    <ToolTip
+      v-if="!disableTooltip"
+      :absolute-left="tooltipLeft"
+      :absolute-top="tooltipTop"
+      class="tooltip"
+      data-testid="sparkline-tooltip"
+      :state="tooltipData"
+      :tooltip-title="tooltipTitle"
+      @dimensions="onTooltipDimensions"
+    />
+
+    <div
+      v-if="hasData && showLabel"
+      class="label"
+      :title="$slots.label ? '' : formattedLabel"
+    >
+      <slot
+        name="label"
+        :total="total"
+        :total-by-dataset="totalByDataset"
+      >
+        {{ formattedLabel }}
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, onUnmounted, reactive, ref, watch } from 'vue'
+import { Bar, Line } from 'vue-chartjs'
+import { format } from 'date-fns'
+import 'chartjs-adapter-date-fns'
+import { type ChartOptions, Tooltip } from 'chart.js'
+import { formatTime } from '@kong-ui-public/analytics-utilities'
+
+import composables from '../composables'
+import { lineChartTooltipBehavior } from '../utils'
+import bucketTimestamps from '../utils/bucketTimestamps'
+import type { ExternalTooltipContext, SparklineDataset, TooltipState } from '../types'
+import { VerticalLinePlugin } from './chart-plugins/VerticalLinePlugin'
+import ToolTip from './chart-plugins/ChartTooltip.vue'
+
+const {
+  datasets,
+  disableTooltip = false,
+  maxCount,
+  maxStamp = Date.now(),
+  minStamp,
+  pointRenderCount = 24,
+  showLabel = false,
+  tooltipTitle,
+  type = 'bar',
+} = defineProps<{
+  datasets: SparklineDataset[]
+  disableTooltip?: boolean
+  /**
+   * determines the relative sizes/heights of bars/points after bucketing
+   */
+  maxCount: number
+  /**
+   * determines the relative position of bars/points, defaults to Date.now()
+   */
+  maxStamp?: number
+  /**
+   * determines the relative position of bars/points.
+   */
+  minStamp: number
+  /**
+   * The number of datapoints to render. Will bucket timestamps into ranges that
+   * are `(maxStamp - minStamp) / pointRenderCount` apart. `pointRenderCount`
+   * must be greater than 1 (will default to 2 if a value less than 2 is
+   * provided) as you can't draw a line without at least 2 points rendered.
+   */
+  pointRenderCount?: number
+  showLabel?: boolean
+  tooltipTitle?: string
+  type: 'bar' | 'line' | 'step'
```

Comment:
If this may become an "official chart type" supported by dashboards/reports, could update these types to be a bit more descriptive. 

EG `spark_line` `spark_bar` `spark_step` ? (kind of following the pattern of timeseries_bar/timeseries_line) 

Could also throw the types [here](https://github.com/Kong/public-ui-components/blob/main/packages/analytics/analytics-chart/src/types/chart-types.ts) where all the types are tracked?